src/SFML/System/FileInputStream.cpp

```diff
@@ -84,13 +86,11 @@ std::optional<std::size_t> FileInputStream::read(void* data, std::size_t size)
 #ifdef SFML_SYSTEM_ANDROID
     if (priv::getActivityStatesPtr() != nullptr)
     {
-        if (!m_androidFile)
-            return std::nullopt;
+        assert(m_androidFile);
```

Comment:
I added a lot of assertions to this file. As far as I'm aware and as far as the tests seem to prove, these assertions never fire and thus they're not necessary. We should be able to remove them. It doesn't seem possible to construct a `FileInputStream` in such a way that we ever try to call a function on a null pointer.

Doesn't hurt much to include them but I'm also of the opinion that I'd rather remove assertions when we can statically prove they're always unnecessary.

Reply:
I'm generally in favour of more assertions, even if they're "unnecessary" they help give local reasoning context when reading the code.

In this particular case they are extremely cheap at run-time as well, so I'd say keep them.

Reply:
Those assertions can fire when with a moved-from FileInputStream.

Reply:
@kimci86: probably it's the 3 hours of sleep, but I'm not seeing the problem. Are you saying these assertions can fire when a moved-from input stream is destroyed? They can definitely fire if a moved-from input stream is used, but it's not meant to be used.

Reply:
Sorry I should have been more explicit.
I am answering to this:
> these assertions never fire and thus they're not necessary
This is not true: the assertions can fire when using a moved-from instance. It is correct to assert because a moved-from instance should not be used. This is an argument in favor of having those asserts.